roboflow/core/version.py

```diff
@@ -535,6 +540,74 @@ def deploy(self, model_type: str, model_path: str) -> None:
                     if file in ["model_artifacts.json", "state_dict.pt"]:
                         raise (ValueError(f"File {file} not found. Please make sure to provide a" " valid model path."))
 
+        self.upload_zip(model_type, model_path)
+
+    def deploy_yolonas(self, model_type: str, model_path: str) -> None:
+        try:
+            import torch
+        except ImportError:
+            raise (
+                "The torch python package is required to deploy yolonas models."
+                " Please install it with `pip install torch`"
+            )
+
+        model = torch.load(os.path.join(model_path, "weights/best.pt"), map_location="cpu")
+        class_names = model["processing_params"]["class_names"]
+
+        opt_path = os.path.join(model_path, "opt.yaml")
+        if not os.path.exists(opt_path):
+            raise RuntimeError(
```

Comment:
this seems like it could be difficult for users to do.. I think many users run this .deploy from inside a colab notebook, do you have a sense for the UX of that? maybe we could add a section to our official notebook that creates the opt.yaml?

Reply:
why would it be difficult for users to do this if they're in a colab? You can write to files with python

but yeah we can add this stuff to the notebook!